cli/src/semgrep/run_scan.py

```diff
@@ -633,13 +633,14 @@ def run_scan(
                     baseline_target_strings = target_strings
                     baseline_target_mode_config = target_mode_config
                     if target_mode_config.is_pro_diff_scan:
+                        scanned = [
```

Comment:
Can we just pass the `scanned` list to the baseline scan? There may be files in there that do not exist in the baseline version. What happens if a file gets renamed, should we scan the same file under the old name?

Reply:
Added a renamed set of files to the target list. Deleted files are checked and removed by `t.exists()` below. A test for removing duplicated findings in renamed files has also been added to the Pro engine PR.